internal/playlist_creator/DirectoryReader.go

```diff
@@ -0,0 +1,64 @@
+package playlist_creator
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"playlistCreator/internal/playlist_creator/config"
+	"sort"
+	"strings"
+)
+
+func ReadFiles(config *config.Config) *FileData {
+	var fileData = FileData{}
+
+	readFilesRecursively(config, config.InputPath, &fileData)
```

Comment:
you should have something like:

```
fileData,err:=readFilesRecursively(config, config.InputPath, &fileData)
```
Then check the error before you proceed to sorting .